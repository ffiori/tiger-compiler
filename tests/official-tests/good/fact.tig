/* define a recursive function */
let
  /* calculate n! */
  function fact (n : int) : int =
    if  n = 0
      then 1
      else n * fact (n - 1)
in
  printInt(fact (10));0 /* Should print 3628800 */
end
