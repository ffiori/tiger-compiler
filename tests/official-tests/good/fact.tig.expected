INT: 3628800
